cartridges/rvw_dev_console/cartridge/controllers/Console.js

```diff
@@ -1,75 +1,104 @@
-'use strict';
-
-const server = require('server');
-const URLUtils = require('dw/web/URLUtils');
-
-const serializer = require('../scripts/serializer');
+"use strict";
 
 /**
  * Display development console template
  */
-server.get(
-    'Show',
-    server.middleware.https,
-    function (req, res, next) {
-        if (dw.system.System.instanceType === dw.system.System.PRODUCTION_SYSTEM) {
-            res.redirect(URLUtils.https('Home-Show').toString());
-
-            return next();
-        }
-
-        res.render('dev_console/index', {
-            urlPath: URLUtils.https('').toString() + '/default',
-            staticPath: URLUtils.staticURL('/').toString()
-        });
-
-        return next();
+function Show() {
+    var ISML = require("dw/template/ISML");
+    var Response = require("dw/system/Response");
+    var System = require("dw/system/System");
+    var URLUtils = require("dw/web/URLUtils");
+
+    response.setHttpHeader(Response.CONTENT_SECURITY_POLICY, "frame-ancestors 'self'");
+    response.setHttpHeader(Response.X_CONTENT_TYPE_OPTIONS, "nosniff");
+
+    if (!request.isHttpSecure()) {
+        response.redirect(URLUtils.https("Console-Show"));
+        return;
+    }
+
+    // We only allow on DEVELOPMENT_SYSTEM and disallow on STAGING_SYSTEM as well
+    // since that is just as sensitive as PROD (due to data replication)
+    if (System.getInstanceType() !== System.DEVELOPMENT_SYSTEM) {
+        response.setStatus(418);
+        response.getWriter().print("I'm a teapot!");
```

Comment:
When you revert the non-staging functionality back, can you replace this message with something that makes sense?

Reply:
done